packages/backend/src/projectAdapters/dbtGitProjectAdapter.ts

```diff
@@ -44,8 +45,18 @@ const gitErrorHandler = (e: Error) => {
             { message: e.message },
         );
     }
+    if (e.message.includes('Repository not found')) {
+        const repo = e.message.match(/github\.com\/([^/]+\/[^/.]+)/)?.[1]; // Get repo from error message
+        throw new NotFoundError(
+            `Could not find Git repository "${repo}". Check that your personal access token has access to the repository and that the repository name is correct.`,
+        );
+    }
     if (e instanceof GitError) {
-        throw new GitError(e.task, stripTokensFromUrls(e.message));
+        throw new UnexpectedGitError(
```

Comment:
Generic error: 

Before, we were throing a `GitError` in some place in the code we check if error is of type LightashError, if not, we shouw a `something went wrong`. THis new error type returns more details to the frontend.